main.tf

```diff
@@ -0,0 +1,162 @@
+########################################################
+### Cloud Trail
+########################################################
+locals {
+  name          = coalesce(var.name, "account-api-trails")
+  s3_key_prefix = coalesce(var.s3_key_prefix, "cloudtrail")
+  bucket_name   = coalesce(var.bucket_name, "cloudtrail-bkt-kdp8784q")
+}
+
+resource "aws_cloudtrail" "main" {
+  name           = "${local.name}-cloudtrail"
+  s3_bucket_name = aws_s3_bucket.cloudtrail.id
+
+  s3_key_prefix                 = var.s3_key_prefix
+  cloud_watch_logs_group_arn    = "${aws_cloudwatch_log_group.cloudtrail.arn}:*" # CloudTrail requires the Log Stream wildcard
+  include_global_service_events = var.include_global_service_events
+  cloud_watch_logs_role_arn     = aws_iam_role.cloudtrail_cloudwatch_role.arn
+  enable_log_file_validation    = var.enable_log_file_validation
+  enable_logging                = var.enable_logging
+  is_multi_region_trail         = var.is_multi_region_trail
+  is_organization_trail         = var.is_organization_trail
+  kms_key_id                    = aws_kms_key.cloudtrail.arn
+  sns_topic_name                = var.sns_topic_name
+
+  ##### Use ONLY ONE of either event_selector or advanced_event_selector
+  ##### Basic event selector resource types, i.e, set using `event_selector` block: `AWS::S3::Object`, `AWS::Lambda::Function`, `AWS::DynamoDB::Table`
+  ##### Basic event selector resource types are valid in advanced event selectors, but advanced event selector resource types are not valid in basic event selectors.
+  ##### https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_DataResource.html
+
+  dynamic "event_selector" {
+    for_each = var.event_selectors
+    content {
+      read_write_type                  = lookup(event_selector.value, "read_write_type", "All")
+      include_management_events        = lookup(event_selector.value, "include_management_events", true)
+      exclude_management_event_sources = lookup(event_selector.value, "exclude_management_event_sources", null)
```

Comment:
add a check for `include_management_events` to be set to  true in order to allow this.